app/services/payment/spam_protection_service.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module Payment
+  class SpamProtectionService
+    MAX_REQUESTS_IN_A_PERIOD = 10
```

Comment:
I think we need to make this more scalable as 10 may end up being a short limit for some enterprise accounts.

Reply:
we are rewriting it to have 10 times in a timespan of 1 hour